src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -229,35 +229,38 @@ void OnOffServer::initOnOffServer(chip::EndpointId endpoint)
         //            set the OnOff attribute to 1.If the previous value of the OnOff
         //            attribute is equal to 1, set the OnOff attribute to 0 (toggle).
         // 0x03-0xfe  These values are reserved.  No action.
-        // 0xff       Set the OnOff attribute to its previous value.
+        // NULL/0xff       Set the OnOff attribute to its previous value.
 
         // Initialize startUpOnOff to No action value 0xFE
-        uint8_t startUpOnOff = 0xFE;
-        EmberAfStatus status = Attributes::StartUpOnOff::Get(endpoint, &startUpOnOff);
+        app::DataModel::Nullable<uint8_t> startUpOnOff;
+        EmberAfStatus status = Attributes::StartUpOnOff::Get(endpoint, startUpOnOff);
         if (status == EMBER_ZCL_STATUS_SUCCESS)
         {
             // Initialise updated value to 0
             bool updatedOnOff = 0;
             status            = Attributes::OnOff::Get(endpoint, &updatedOnOff);
             if (status == EMBER_ZCL_STATUS_SUCCESS)
             {
-                switch (startUpOnOff)
+                if (!startUpOnOff.IsNull())
                 {
-                case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_OFF:
-                    updatedOnOff = 0; // Off
-                    break;
-                case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_ON:
-                    updatedOnOff = 1; // On
-                    break;
-                case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_TOGGLE:
-                    updatedOnOff = !updatedOnOff;
-                    break;
-                case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_PREVIOUS:
-                default:
-                    // All other values 0x03- 0xFE are reserved - no action.
-                    // When value is 0xFF - update with last value - that is as good as
-                    // no action.
-                    break;
+                    switch (startUpOnOff.Value())
+                    {
+                    case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_OFF:
+                        updatedOnOff = 0; // Off
+                        break;
+                    case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_ON:
+                        updatedOnOff = 1; // On
+                        break;
+                    case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_TOGGLE:
+                        updatedOnOff = !updatedOnOff;
+                        break;
+                    case EMBER_ZCL_START_UP_ON_OFF_VALUE_SET_TO_PREVIOUS:
```

Comment:
This case is effectively equivalent to startUpOnOff being null right? If so, it shouldn't be inside this if() branch..

Reply:
It has no logic effect here since it's a NOP, but still worthwhile removing it to make the logic look aligned with the spec.

Reply:
You are right.  Removed that case.